Source/WebCore/dom/SimpleRange.h

```diff
@@ -29,6 +29,47 @@
 
 namespace WebCore {
 
+struct SimpleRangeWeak {
+    BoundaryPointWeak start;
+    BoundaryPointWeak end;
+
+    std::optional<std::reference_wrapper<Node>> startContainer() const
+    {
+       if (!start.container)
+            return std::nullopt;
+        return *start.container;
+    }
+
+    std::optional<Ref<Node>> protectedStartContainer() const
+    {
+        if (!start.container) {
+            return std::nullopt;
+        }
+        return Ref { *start.container };
+    }
+
+    unsigned startOffset() const { return start.offset; }
+
+    std::optional<std::reference_wrapper<Node>> endContainer() const
+    { 
+       return startContainer();
+    }
+
+    std::optional<Ref<Node>> protectedEndContainer() const
+    {
+        return protectedStartContainer();
+    }
+    unsigned endOffset() const { return end.offset; }
+
+    bool collapsed() const { return start == end; }
+
+    friend bool operator==(const SimpleRangeWeak&, const SimpleRangeWeak&) = default;
```

Comment:
These methods are not actually used by the only instance of SimpleRangeWeak which is in m_dragStartSelection. SimpleRange had these functions implemented, and I was unsure if I should just fill out this functionality now or leave it for the future, where these methods would be implemented on an as-needed basis.

Reply:
I think we should not have them if they're unused. They're also kind of awkward. An `optional<Ref>` behaves functionally like a `RefPtr` in that either it has a value or it doesn't. Same with `optional<std::reference_wrapper<Node>>` that's just a `Node*`.

If we don't need the functions then I don't think we should include them in this patch.

Reply:
Done.